### PR TITLE
[INLONG-10395][Manager] Add interface for schedule client and engine

### LINKED ISSUE #10395: [Feature][Manager] Add interface for schedule client and engine
### Description

We want to support periodic scheduling capability in the InLong manager, mainly including two parts.
1. add interfaces for schedule client and engine
2. add basic schedule capability for schedule client and engine based on quartz

this is the first part, adding the interfaces for schedule client and engine

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by aloyszhang**
Fixes #10395 

### Motivation

The first part of support schedule client and engine, define the interfaces for schedule client and engine

### Modifications

1. create a new module `manager-schedule` in manager 
2. add interfaces for scheduler client and engine

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngine.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * */
+public interface ScheduleEngine {
+
+    /**
+     * Start schedule engine.
+     * */
+    void start();
+
+    /**
+     * handle schedule register.
```
Comment (fuweng11) [inline inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngine.java]: Handle schedule register.

File: inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngine.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     * @param scheduleInfo schedule info to register
+     * */
+    boolean handleRegister(ScheduleInfo scheduleInfo);
+
+    /**
+     * handle schedule unregister.
```
Comment (fuweng11) [inline inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngine.java]: Handle schedule register.

File: inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngine.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * @param scheduleInfo schedule info to unregister
+     * */
+    boolean handleUnregister(ScheduleInfo scheduleInfo);
+
+    /**
+     * handle schedule update.
```
Comment (fuweng11) [inline inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngine.java]: Handle schedule update.

File: inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngineClient.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    boolean register(ScheduleInfo scheduleInfo);
+
+    /**
+     * unregister schedule from schedule engine.
+     * @param scheduleInfo schedule info to unregister
+     * */
+    boolean unregister(ScheduleInfo scheduleInfo);
+
+    /**
+     * update schedule from schedule engine.
```
Comment (fuweng11) [inline inlong-manager/manager-schedule/src/main/java/org/apache/inlong/schedule/ScheduleEngineClient.java]: Ditto.

**Review by fuweng11 — APPROVED** [review]

**Review by baomingyu — APPROVED** [review]
+1

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**